sdk/core/core-rest-pipeline/CHANGELOG.md

```diff
@@ -2,6 +2,13 @@
 
 ## 1.0.4 (Unreleased)
 
+- A new type is exported, `ChallengeCallbackOptions`, which contains `scopes`, a string `claims` that might contain a challenge, the previous token used on a request `previousToken`, a `getToken` function, which calls to the underlying credential's `getToken` with a smart approach that minimizes unnecessary network requests, the pipeline request `request`, and a `setAuthorizationHeader` which makes it easier to set the token on the pipeline request.
+- Added a `challengeCallbacks` optional property to the `bearerTokenAuthenticationPolicy` that allows it to process authentication challenges, as follows:
```

Comment:
@jeremymeng hi hi! Since this is in a new policy we should change this (and probably the PR description too, since I copied the changelog entries there)

Reply:
absolutely!  Will re-write once things are settled down

Reply:
Rewrote PR description and CHANGELOG